File: app/src/main/java/com/ndmsystems/coala/layers/ReceiveLayer.java
package com.ndmsystems.coala.layers;

import com.ndmsystems.coala.message.CoAPMessage;
import com.ndmsystems.coala.utils.Reference;

import java.net.InetSocketAddress;

public interface ReceiveLayer {
    boolean onReceive(CoAPMessage message, Reference<InetSocketAddress> senderAddress);
}

File: app/src/main/java/com/ndmsystems/coala/layers/ReceiveLayer.kt
package com.ndmsystems.coala.layers

import com.ndmsystems.coala.LayersStack
import com.ndmsystems.coala.message.CoAPMessage
import com.ndmsystems.coala.utils.Reference
import java.net.InetSocketAddress

interface ReceiveLayer {
    fun onReceive(message: CoAPMessage, senderAddressReference: Reference<InetSocketAddress>): LayersStack.LayerResult
}